cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -536,6 +539,41 @@ class PyPrimitiveConverter<T, enable_if_binary<T>>
     return Status::OK();
   }
 
+  Result<int64_t> ExtendAsMuchAsPossible(PyObject* values, int64_t size,
+                                         int64_t offset) override {
+    DCHECK_GE(size, offset);
+    // See BaseBinaryBuilder::Resize - avoid error in Reserve()
+    ARROW_LOG(WARNING) << "size=" << size << " offset=" << offset
```

Comment:
Did you mean to check these log statements in?

Reply:
Whoops, didn't mean to reveal my amazing debugging secrets :) (thanks for catching this)